lesson10/index.js

```diff
@@ -0,0 +1,61 @@
+const ul = document.getElementById('js-lists');
+
+const listData = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+function displayLoadingImage () {
+    const loadingImage = document.createElement('img');
+    loadingImage.id = 'loadingImageId';
+    loadingImage.src = './loading-circle.gif';
+    ul.appendChild(loadingImage);
+}
+
+function removeLoadingImage () {
+    const loadingImage = document.getElementById('loadingImageId');
+    loadingImage.remove();
+}
+
+async function getListData() {
+    return new Promise(resolve => {
+        displayLoadingImage ();
+        setTimeout(() => {
+            resolve(listData);
+        }, 3000);
+    });
+}
+
+async function tryGetData() {
+    try {
+        return await getListData();
+    } catch (error) {
+        const errorText = document.createElement('p');
+        ul.replaceWith(errorText);
+        errorText.textContent = "エラーが発生しました";
+    } finally {
+        removeLoadingImage ();
+    }
+}
```

Comment:
[codesandbox](https://codesandbox.io/s/kong-nopei-lie-gadu-tutekitachang-he-hua-mian-nihe-mobiao-shi-sarenaidesu-664oq?file=/index.js)
With the current code, even if an empty array is passed, the process will terminate normally.

It's rough, but IMO.
```suggestion
async function tryGetData() {
  try {
    const value = await getListData();
    if (value.length) {
      return value;
    } else {
      const errorText = document.createElement("p");
      ul.replaceWith(errorText);
      errorText.textContent = "値が入ってません";
    }
  } catch (error) {
    const errorText = document.createElement("p");
    ul.replaceWith(errorText);
    errorText.textContent = "エラーが発生しました";
  } finally {
    removeLoadingImage();
  }
}
```

Reply:
IMO @443n0511 

1. Instead of judging the value there, I felt it was good to say that if there was an error, it would be returned as it is. Because the empty Array is also the result of success.
2. I want to avoid getting another error in the try and getting confused.
3. I feel like a function called tryGetData can return an error.

What do you think.

https://codesandbox.io/s/kong-nopei-lie-gadu-tutekitachang-he-hua-mian-nihe-mobiao-shi-sarenaidesu-forked-mo6jn?file=/index.js

Reply:
I let this function do too much work. You are right, thank you for pointing that out.`tryGetData` simply needs to get the value.

@sae-github I'm sorry for the confusion. Morita-san's IMO is also my IMO.
Can you check Morita-san's sandbox?

Reply:
@443n0511 
Thank you for your reviewing!
I found the problem with your point.

- I changed the name `tryGetData` to `getData`.
- I created the function `errorText`.
- I changed to display an error text when it is an empty array.

I fixed.
-> 29aeee5

@kenmori 
Thank you for checking the assignment.
I have a hard time coming up with a proper function name.
Thank you for pointing out.